config/config.go

```diff
@@ -530,10 +532,69 @@ type AuthConfig struct {
 	TLSAuthEntities []string            `json:"tls_auth_entities"`
 }
 
+// WebOauthConfig contains the structued AuthHandlerConfig for the CredentialsTypeOAuthWeb type.
+type WebOauthConfig struct {
+	// The allowed jwt audiences the OAuthWeb auth handler will accept.
+	AllowedAudiences []string `json:"allowed_audiences"`
+	// contains the raw jwks json.
+	JSONKeySet AttributeMap `json:"jwks"`
+
+	// on validation the JSONKeySet is validated and parsed into this field and can be used.
+	ValidatedKeySet jwks.KeySet `json:"-"`
+}
+
+// Validate returns true if the WebOAuth is valid. Ensures each key is valid and meets the required constraints.
+func (c *WebOauthConfig) Validate(path string) error {
+	jsonJWKs, err := json.Marshal(c.JSONKeySet)
+	if err != nil {
+		return utils.NewConfigValidationError(fmt.Sprintf("%s.config", path), errors.Wrap(err, "failed to marshal jwks"))
```

Comment:
I think it's `jwks`, not config for both errors here

Reply:
Done.